### DISCUSSION TITLE
My implement for mjai.app for kanachan

### DISCUSSION THREAD

**Discussion by shinkuan** [Cryolite/kanachan#50] [Show and tell]
### Code
[shinkuan/kanachan.mjai.app](https://github.com/shinkuan/kanachan.mjai.app)
Based on [mjai.app](https://github.com/Cryolite/kanachan/tree/main/mjai.app) in this repository.

### Change
- Remove the need for [xiangting_calculator](https://github.com/Cryolite/kanachan/tree/main/mjai.app/xiangting_calculator)
- Use mahjong.shanten instead
- Fix a bug where jiagang value is wrong https://github.com/shinkuan/kanachan.mjai.app/commit/38263ba42fc2f5ddcde5f721ec8df3602ca5144f
- Fix other small bugs

### Log
Simulate log of four kanachan mjai bot: [mjai.log](https://github.com/Cryolite/kanachan/files/12327410/mjai.log)

Feel free to ask me if you encounter any problem using this.

https://github.com/Cryolite/kanachan/issues/20

**Comment by shinkuan** [comment]
### Update
- 立直中の送り槓を禁止する．https://github.com/shinkuan/kanachan.mjai.app/commit/348143e6ce0fd598511a4c306603fe7924fc88c9
- 河底牌に対する大明槓が可能かどうか確認する．https://github.com/shinkuan/kanachan.mjai.app/commit/65c08a97b7aa332a6043b5acce498fdc798f7448
- 暗槓に対する国士無双の槍槓をチェックする．https://github.com/shinkuan/kanachan.mjai.app/commit/0cba7cbc6cf225245e35aba143a0c5a9754c5a7f

**Comment by Cryolite** [comment]
I greatly appreciate your feedback. With the modifications, including your feedback, I have been able to officially start supporting [AIJansou (mjai.app)](https://mjai.app/) in the `develop` branch.
